### PR TITLE
feat(route): add Yoasobi Official support

### PR DESCRIPTION
**Pull request by Kiotlin**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/yoasobi-music/info/news
/yoasobi-music/info/biography
/yoasobi-music/live
/yoasobi-music/media
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [x] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
Yoasobi, a Japanese music [superduo](https://en.wikipedia.org/wiki/Superduo) formed by Sony Music. Add new routes for its [official website](https://www.yoasobi-music.jp/).

### CODE REVIEW

File: lib/v2/yoasobi-music/info.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            // the article content
+            description: art(path.join(__dirname, 'templates/info.art'), {
+                image: i.image,
+                category: i.category,
+                description: i.description.replace(/\n/g, '<br>'),
+            }),
+            // the article publish time
+            pubDate: parseDate(i.date),
+            // the article link
+            link: i.id ? `${officialUrl}/${i.id}` : officialUrl,
```
Comment (TonyRL) [inline lib/v2/yoasobi-music/info.js:61]: link is also used as `guid` if not specified, which should be unique

File: lib/v2/yoasobi-music/templates/live.art [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{{ if image }}
+    <img src={{image}} />
+{{ /if }}
+{{@ description }}
+{{ each sessions se i}}
```
Comment (TonyRL) [inline lib/v2/yoasobi-music/templates/live.art:5]:
```suggestion
{{ each sessions se }}
```
You can omit `i` if it isn't used

File: lib/v2/yoasobi-music/media.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            // the article title
+            title: i.title,
+            // the article content
+            description: art(path.join(__dirname, 'templates/media.art'), {
+                date: i.date,
+                weekDay: i.weekDay,
+                postFix: i.startTime && i.endTime && i.tvStation ? `${i.startTime} ~ ${i.endTime} ${i.tvStation}` : null,
+                description: i.description,
+            }),
+            // no pubDate
```
Comment (TonyRL) [inline lib/v2/yoasobi-music/media.js:51]: `i.date` can be used as pubDate

File: lib/v2/yoasobi-music/media.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            // the article link
+            link: officialUrl,
```
Comment (TonyRL) [inline lib/v2/yoasobi-music/media.js:53]: link is also used as `guid` if not specified, which should be unique

File: lib/v2/yoasobi-music/media.js (hunk 3 above)
Comment (Kiotlin) [reply lib/v2/yoasobi-music/media.js:51]: Actually the `i.date` refers to the air time of a tv/radio/... show, which has a different meaning with the date a post being published. Is there any side effect if I left `pubDate` param empty?

File: lib/v2/yoasobi-music/info.js (hunk 1 above)
Comment (Kiotlin) [reply lib/v2/yoasobi-music/info.js:61]: Same with [media.js](https://github.com/DIYgod/RSSHub/pull/10920#discussion_r981154009). Every news have their own unique page url `${officialUrl}/${i.id}`, but when comes to `biography`, they become pure `<div>` embeded in the page. [news](https://www.yoasobi-music.jp/news) [biography](https://www.yoasobi-music.jp/biography)

File: lib/v2/yoasobi-music/media.js (hunk 4 above)
Comment (TonyRL) [reply lib/v2/yoasobi-music/media.js:53]: explicitly set `guid` to something like `guid: i.title + i.date`

File: lib/v2/yoasobi-music/media.js (hunk 3 above)
Comment (TonyRL) [reply lib/v2/yoasobi-music/media.js:51]: You can leave `pubDate` empty and the feed will still work. But the user side, users cannot sort their feeds by date if it isn't provided

Comment (Kiotlin) [reply lib/v2/yoasobi-music/media.js:51]: OK, I see. I will use `i.date` as pubDate

**Pull request merged**